### ISSUE TITLE
Established Connection Issue 

### ISSUE THREAD

**Issue by shayan555** [voximplant/react-native-demo#83]
Hi All,

We are facing issue once the connection disconnected then not able to reestablished the connection and showing error 'Temporarily Unavailable'

**Comment by YuliaGrigorieva** [comment]
Hello @shayan555 ,

Do you mean that the callee does not receive the incoming call and the caller gets "Temporary unavailble" error from CallFailed event?

Best regards,
Yulia Grigorieva

**Comment by shayan555** [comment]
Log : Client: emit event ConnectionClosed

Once the connection closed then callee does not receive the incoming call and the caller gets "Temporary unavailble" error from CallFailed

**Comment by ahammedmishal** [comment]
yes i got the same issue after some time after login when we call the user it will show Call failed : 'Temporarily Unavailable'

**Comment by wilmxre** [comment]
@ahammedmishal did you manage to fix this issue?

**Comment by ahammedmishal** [comment]
> @ahammedmishal did you manage to fix this issue?

Yes i used direct google translation api

**Comment by ahammedmishal** [comment]
@ahammedmishal did you manage to fix this issue?

there is no fix by them so do your own logic

**Comment by wilmxre** [comment]
@ahammedmishal how did you fix it?

**Comment by ahammedmishal** [comment]
in root file of the app i do like this . useEffect(() => {
    testConnection = async () => {
      let state = await Voximplant.getInstance().getClientState();      
      if (state !== Voximplant.ClientState.DISCONNECTED) {        
        loginWithToken();
      } else {
        await Voximplant.getInstance().connect();
      }
    };
    const interval = setInterval(testConnection, 5000);
    return () => clearInterval(interval);
  }, []);

  loginWithToken = async () => {
    try {
      let state = await Voximplant.getInstance().getClientState();
      if (state === Voximplant.ClientState.DISCONNECTED) {
        await Voximplant.getInstance().connect();
      }
      if (state !== Voximplant.ClientState.LOGGED_IN) {
        let usernames = `${userFullDetails?.username}@sdk-tutorial-69p5a6q.bhagyavakta`;
        let passwords = `${userFullDetails?.uid}`;
        const loginClicked = async (usernames, passwords) => {
          await AsyncStorage.setItem('usernameValue', usernames);
          await AsyncStorage.setItem('uid', passwords);
          LoginManager.getInstance().loginWithPassword(
            usernames + '.voximplant.com',
            passwords,
          );
        };
        loginClicked(usernames, passwords);
      }
    } catch (e) {
      console.log('LoginManager: loginWithToken: ' + e.name);
      if (e.name === Voximplant.ClientEvents.AuthResult) {
        console.log('LoginManager: loginWithToken: error code: ' + e.code);
      }
    }
  };
